lib/postprocessing/image.js

```diff
@@ -178,6 +181,35 @@ async function needsImagePostProcess(rendition) {
     return false;
 }
 
+async function processWatermark(img, renditionInstructions, outdir) {
+    console.log('Processing watermark...');
+
+    const watermarkProcessor = new WatermarkProcessor(renditionInstructions);
+    const resizedWatermarkPath = renditionInstructions.watermark.name ? path.join(outdir, 'resized-'+ renditionInstructions.watermark.name)
+        : path.join(outdir, 'resized-'+ path.basename(renditionInstructions.watermark.path));
+
+    try {
+        await watermarkProcessor.fitWatermark(resizedWatermarkPath);
+    } catch(err) {
+        console.error('Error while resizing watermark to rendition dimensions', err);
+        // throw err; or should we revert to using original watermark as is i.e use renditionInstructions.watermark.path ?
```

Comment:
I'd suggest to throw:
If you have a very large watermark, you may actually end up being in a "transparent" part of the watermark and it would look like there is no watermark at all.

Reply:
+1 for throwing an error